File: Creational/FactoryMethod/PHP/factories/AbstractWindow.php
<?php

require_once(__DIR__.'/../interfaces/IButton.php');

/**
 * Abstract factory.
 */

abstract class AbstractWindow
{
	public function paint()
	{
		print("Painting window.\n");

		$button = $this->createButton();

		$button->setPosition(1, 2);
		$button->paint();
	}

	protected abstract function createButton(): IButton;
}

File: Creational/FactoryMethod/Java/factories/AbstractWindow.java
package patterns.creational.factorymethod.factories;

import patterns.creational.factorymethod.interfaces.IButton;

/**
 * Abstract factory.
 */

public abstract class AbstractWindow
{
	public void paint()
	{
		System.out.println("Painting window.");

		IButton button = this.createButton();

		button.setPosition(1, 2);
		button.paint();
	}
	
	protected abstract IButton createButton();
}
